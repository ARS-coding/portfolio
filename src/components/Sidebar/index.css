.sidebar {
    bottom: 0;
    width: max(5%, 50px);
    font-size: 0.5rem;
    position: fixed;
}

.icon {
    color: white;
    margin-top: 30px;
    margin-bottom: 30px;
    padding: 10px;
    cursor: pointer;
}

@media only screen and (max-width: 575px) {
    .sidebar {
        padding: 1%;
    }
}

@media only screen and (max-width: 300px) {
    .sidebar {
        display: none !important;
    }
}

.fa-phone {
    color: white;
}

.notification {
    color: white;
}

.iconContainer {
    margin: auto;
}
