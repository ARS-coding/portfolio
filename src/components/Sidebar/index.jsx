import React from 'react';
import './index.css';
import { Popover, OverlayTrigger } from 'react-bootstrap';

const Sidebar = () => {
    const phonePopover = (
        <Popover id='popover-basic'>
            <Popover.Content>
                <span>+90 552 360 08 93 </span>
            </Popover.Content>
        </Popover>
    );

    const emailPopover = (
        <Popover id='popover-basic'>
            <Popover.Content>
                <span>tarkanmansuroglu@hotmail.com</span>
            </Popover.Content>
        </Popover>
    );
    return (
        <div className='sidebar float-left d-flex flex-column-reverse'>
            <div className='iconContainer'>
                <a
                    target='_blank'
                    rel='noreferrer'
                    href='https://www.instagram.com/tarkanmansuroglu/'
                >
                    <i class='fab fa-instagram icon fa-3x'></i>
                </a>
            </div>
            <div className='iconContainer'>
                <a
                    target='_blank'
                    rel='noreferrer'
                    href='https://www.twitter.com'
                >
                    <i class='fab fa-twitter icon fa-3x'></i>
                </a>
            </div>
            <div className='iconContainer'>
                <a
                    target='_blank'
                    rel='noreferrer'
                    href='https://github.com/tmansuroglu'
                >
                    <i class='fab fa-github icon fa-3x'></i>
                </a>
            </div>
            <div className='iconContainer'>
                <a
                    target='_blank'
                    rel='noreferrer'
                    href='https://www.linkedin.com/in/tarkan-mansuro%C4%9Flu-3385301bb/'
                >
                    <i class='fab fa-linkedin-in icon fa-3x'></i>
                </a>
            </div>
            <div className='iconContainer'>
                <OverlayTrigger
                    trigger='click'
                    placement='right'
                    overlay={emailPopover}
                >
                    <i class='fas fa-envelope-open icon fa-3x'></i>
                </OverlayTrigger>
            </div>
            <div className='iconContainer'>
                <OverlayTrigger
                    trigger='click'
                    placement='right'
                    overlay={phonePopover}
                >
                    <i class='fas fa-phone fa-3x icon'></i>
                </OverlayTrigger>
            </div>
        </div>
    );
};

export default Sidebar;
