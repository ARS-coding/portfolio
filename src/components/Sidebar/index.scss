@import '../../assets/variables';

.sidebar {
  bottom: 0;
  font-size: 0.5rem;
  position: fixed;
  left: 0;
  width: 60px;
  a {
    text-decoration: none;
    &:hover {
      text-decoration: none;
    }
  }
}

.icon {
  color: white;
  cursor: pointer;
  padding: 10px 13px;
  margin: 15px 0;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 80%;
  text-decoration: none;
}

.iconContainer {
  width: 100%;
  height: 100%;

  div {
    width: 100%;
    height: 100%;
  }
}

@media only screen and (max-width: $sm) {
  .sidebar {
    display: none;
  }
}
