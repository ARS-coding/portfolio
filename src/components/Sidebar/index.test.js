import React from 'react';
import renderer from 'react-test-renderer';
import Sidebar from '.';

it('Sidebar renders correctly', () => {
    const tree = renderer.create(<Sidebar />).toJSON();
    expect(tree).toMatchSnapshot();
});
