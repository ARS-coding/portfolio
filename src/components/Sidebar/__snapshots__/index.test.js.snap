// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  className="sidebar"
>
  <div
    className="iconContainer"
  >
    <i
      className="fas fa-phone fa-3x icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="iconContainer"
  >
    <i
      className="fas fa-envelope-open icon fa-3x"
      onClick={[Function]}
    />
  </div>
  <div
    className="iconContainer"
  >
    <a
      href="https://www.linkedin.com/in/tarkanmansuroglu/"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-linkedin-in icon fa-3x"
      />
    </a>
  </div>
  <div
    className="iconContainer"
  >
    <a
      href="https://github.com/tmansuroglu"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-github icon fa-3x"
      />
    </a>
  </div>
  <div
    className="iconContainer"
  >
    <a
      href="https://www.instagram.com/tarkanmansuroglu/"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-instagram icon fa-3x"
      />
    </a>
  </div>
  <div
    className="iconContainer"
  >
    <a
      href="https://twitter.com/T_Mansuroglu"
      rel="noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-twitter icon fa-3x"
      />
    </a>
  </div>
</div>
`;
